app-modules/engagement/src/Filament/Actions/RelationManagerSendEngagementAction.php

```diff
@@ -81,14 +86,70 @@ protected function setUp(): void
-                    ->default(NotificationChannel::Email->value)
-                    ->disableOptionWhen(fn (RelationManager $livewire, string $value): bool => (($value == (NotificationChannel::Sms->value) && ! $livewire->getOwnerRecord()->canReceiveSms())) || NotificationChannel::tryFrom($value)?->getCaseDisabled())
-                    ->selectablePlaceholder(false)
-                    ->live(),
+            ->mountUsing(function (array $arguments, Form $form, RelationManager $livewire) {
+                $livewire->dispatch('engage-action-finished-loading');
+
+                if (filled($arguments['route'])) {
+                    $form->fill([
+                        'channel' => $arguments['channel'] ?? 'email',
+                        'recipient_route_id' => $arguments['route'],
+                        'signature' => auth()->user()->signature,
+                    ]);
+                } else {
+                    $form->fill();
+                }
+            })
+            ->form(fn (Form $form, RelationManager $livewire) => $form->schema([
+                Grid::make(2)
+                    ->schema([
+                        Select::make('channel')
+                            ->label('What would you like to send?')
+                            ->options(NotificationChannel::getEngagementOptions())
+                            ->default(NotificationChannel::Email->value)
+                            ->disableOptionWhen(fn (RelationManager $livewire, string $value): bool => (($value == (NotificationChannel::Sms->value) && (! $livewire->getOwnerRecord()->phoneNumbers()->where('can_receive_sms', true)->exists()))) || NotificationChannel::tryFrom($value)?->getCaseDisabled())
+                            ->selectablePlaceholder(false)
+                            ->live()
+                            ->afterStateUpdated(function (mixed $state, RelationManager $livewire, Set $set) {
+                                $channel = NotificationChannel::parse($state);
+
+                                $route = match ($channel) {
+                                    NotificationChannel::Email => $livewire->getOwnerRecord()->primaryEmailAddress?->getKey(),
+                                    NotificationChannel::Sms => $livewire->getOwnerRecord()->primaryPhoneNumber()
+                                        ->where('can_receive_sms', true)
+                                        ->first()?->getKey(),
+                                } ?? match ($channel) {
+                                    NotificationChannel::Email => $livewire->getOwnerRecord()->emailAddresses()
+                                        ->first()?->getKey(),
+                                    NotificationChannel::Sms => $livewire->getOwnerRecord()->phoneNumbers()
+                                        ->where('can_receive_sms', true)
+                                        ->first()?->getKey(),
+                                };
```

Comment:
Why the seperate `match` statements here? Why not inline the null check in each?

Reply:
IMO it looks cleaner this way, it's two distinct levels of checking to prevent a very long match arm which executes 2 queries